main.py

```diff
@@ -491,20 +491,24 @@ def get_redirect_route(
         r'/<subtopic_url_fragment>' % feconf.SUBTOPIC_DATA_HANDLER,
         subtopic_viewer.SubtopicPageDataHandler),
     get_redirect_route(
-        r'%s/revision' % feconf.TOPIC_VIEWER_URL_PREFIX,
-        topic_viewer.TopicViewerPage),
+        r'%s/revision' % feconf.ACCESS_VALIDATION_HANDLER_PREFIX,
+        access_validators.TopicViewerPageAccessValidationHandler),
     get_redirect_route(
         r'%s/revision/<subtopic_url_fragment>' %
         feconf.TOPIC_VIEWER_URL_PREFIX, subtopic_viewer.SubtopicViewerPage),
     get_redirect_route(
         r'%s/<topic_id>' % feconf.TOPIC_EDITOR_STORY_URL,
         topic_editor.TopicEditorStoryHandler),
+
     get_redirect_route(
-        r'%s' % feconf.TOPIC_VIEWER_URL_PREFIX,
-        topic_viewer.TopicViewerPage),
+        r'%s/can_access_topic_viewer_page' %
+        feconf.ACCESS_VALIDATION_HANDLER_PREFIX,
+        access_validators.TopicViewerPageAccessValidationHandler
+    ),
+
     get_redirect_route(
-        r'%s/practice' % feconf.TOPIC_VIEWER_URL_PREFIX,
-        topic_viewer.TopicViewerPage),
+        r'%s/practice' % feconf.ACCESS_VALIDATION_HANDLER_PREFIX,
+        access_validators.TopicViewerPageAccessValidationHandler),
```

Comment:
Why does/revision || /practice need to be handled via Acces validator?

Reply:
as both the routes are dependent on `TopicViewerPage` and we deleted that, so i put it through the access validator

Reply:
if both use same access validator, then can't we have one API? (instead of two separate API)

Reply:
I agree with @DubeySandeep, please reuse the access validator.